### PR TITLE
Enhancement dependency reduction crosschain

### PR DESCRIPTION
**Pull request by gldeng**
- [x] Dependency reduction for cross chain module.
- [x] Redefine components and structures.
- [x] Improve cross chain communication.
- [x] Support main/side chain node restart.
- [x] Avoid indexing redundant parent chain data.

### CODE REVIEW

File: src/AElf.CrossChain.Core/CrossChainConsts.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+namespace AElf.CrossChain
+{
+    internal class CrossChainConsts
```
Comment (rosona) [inline src/AElf.CrossChain.Core/CrossChainConsts.cs]: change *Consts to *Constants

Comment (rosona) [reply src/AElf.CrossChain.Core/CrossChainConsts.cs]: Remove unused CrossChainConsts: AElf.Contract.CrossChain.Tests CrossChainConsts

Comment (rosona) [reply src/AElf.CrossChain.Core/CrossChainConsts.cs]: remove unused : internal class CrossChainConsts

**Comment by loning** [comment]
@shiwk pls commit changes request

**Pull request merged**